src/Ads/AssetSuggestionsService.php

```diff
@@ -97,21 +146,20 @@ protected function get_post_assets( int $id ): array {
 			$attachments_ids = [ ...$attachments_ids, ...$product->get_gallery_image_ids(), $product->get_image_id() ];
 		}
 
-		$gallery_images_urls = get_post_gallery_images( $id );
-		$marketing_images    = [ ...$this->get_url_attachments_by_ids( $attachments_ids ), ...$gallery_images_urls ];
-		$marketing_images    = array_slice( $marketing_images, 0, self::DEFAULT_MAXIMUM_MARKETING_IMAGES );
-		$long_headline       = get_bloginfo( 'name' ) . ': ' . $post->post_title;
+		$attachments_ids  = array_slice( [ ...$attachments_ids, ...$this->get_gallery_images_ids( $id ) ], 0, self::DEFAULT_MAXIMUM_MARKETING_IMAGES );
+		$marketing_images = $this->get_url_attachments_by_ids( $attachments_ids );
```

Comment:
💅 

Maybe we can add the `array_slice` logic in the `get_url_attachments_by_ids` scope. 

I see we do some preparation inside `get_url_attachments_by_ids` so maybe we can do something like :

```php
function prepare_image_ids( $ids ) {
  $ids = array_unique( ArrayUtil::remove_empty_values( $ids ) );
  $ids = array_map( 'intval', $ids );
  return [$ids, 0, self::DEFAULT_MAXIMUM_MARKETING_IMAGES];
}

[...]

protected function get_url_attachments_by_ids( $ids, ... ) {
   $ids = $this->prepare_image_ids($ids);
   [...] 
}
```

Reply:
I added your suggestion with one extra argument `$maximum_images` so we can make this function more flexible.
See: https://github.com/woocommerce/google-listings-and-ads/pull/1804/commits/9677071533e50f43fb89beafc6b6d12731089fbd